Multiple instance events anim develop

ChrisEberleSchool opened the pull request:
Fixed the issue where you were able to open multiple instances of the same event. This change still allows you to open multiple instances of events so long as they are not the same.

BEFORE:
![image](https://github.com/xspanger3770/GlobalQuake/assets/156865868/7d0d0218-bb23-4a6f-8971-bddfa1393f61)

Also this same thing was occurring in the event panel in the animation button where multiple same instance animations could be open at once which I fixed!

ChrisEberleSchool committed:
- animation instances fix

fixed it so only one instance of the animation panel can be open from a single event instance

ChrisEberleSchool committed:
- fixed issue

when u closed a instance of animation you couldn't reopen it again

ChrisEberleSchool committed:
- multiple instanced events fixed

- this change ensures you can still open multiple event instances so long as they are different event. If you try opening similiar events they will not open.

ChrisEberleSchool committed:
- closing flag reset

ChrisEberleSchool committed:
- all issues resolved!

JABirchall commented:
One bug i can think of with this is, wouldnt the corrisponding earthquakes index always be changing as new earthquakes are added and old ones purged?

ArchivedQuakes has a UUID property on it, it would be best to use that to track open windows and animations, the current implimentation is basically a race condition.

xspanger3770 committed:
Merge pull request #252 from xspanger3770/develop

Finally update main branch with 0.11.x version features (336 commits) so that my github profile page lights up like a christmas tree and I can start updating the readme files

xspanger3770 committed:
Merge pull request #253 from xspanger3770/main

Update readme branch

xspanger3770 committed:
update general info in README.md

xspanger3770 committed:
Merge pull request #254 from xspanger3770/readme

Readme

ChrisEberleSchool committed:
Merge branch 'xspanger3770:main' into multiple-instance-events-anim-develop

ChrisEberleSchool committed:
- update branch to get ready for new changes

reverts to standard

ChrisEberleSchool committed:
revert readme

ChrisEberleSchool committed:
- using UUID of archived quakes to handle instances

- I am creating an arrayList of UUID's called openedQuakes. Before a panel can be open its uuid must not be in the array list, checking this with a new  isOpened comparison boolean method.

ChrisEberleSchool committed:
- BugFix final commit

- this commit handles the animation panel multiple instances.

ChrisEberleSchool committed:
- removed unneccessary imports

ChrisEberleSchool commented:
I fully redid the bugfix in a much simpler way using UUID's as @JABirchall suggested. Should work perfectly now

JABirchall commented on GlobalQuakeClient/src/main/java/globalquake/ui/globalquake/EarthquakeListPanel.java:
```diff
@@ -109,6 +138,23 @@ public void mouseMoved(MouseEvent e) {
+        // loop through openedquakes
+        for( UUID i: openedQuakes) {
+            if( i == id) {
+                return true;
```
Would it be possible to use the ArrayList builtin `indexOf(id)` or `contains(id)` instead of looping over the array?

ChrisEberleSchool replied:
Just replaced my isOpened method with contains! Thanks for the tip, didn't know this method existed!!

JABirchall commented on GlobalQuakeClient/src/main/java/globalquake/ui/globalquake/EarthquakeListPanel.java:
```diff
@@ -77,15 +87,34 @@ public void mousePressed(MouseEvent e) {
+                                openedQuakes.remove(quakeUuid);
+                            }
+                            public void windowClosed(WindowEvent e) {
+                                openedQuakes.remove(quakeUuid);
```
Do we need to remove the uuid twice on closing and closed event? 
is there a case where 1 of these events doesnt do what it is supposed to?

ChrisEberleSchool replied:
This is pretty much for redundancy, I did research and I found it is best to include both!

JABirchall replied:
All good, just wanted to know the reason around it.

ChrisEberleSchool committed:
- using contains method instead of isOpened method

JABirchall commented:
This is ready for @xspanger3770 review.

xspanger3770 commented:
https://github.com/xspanger3770/GlobalQuake/tree/announcement
